components/Headlines.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from "react";
+import { useChannel } from "@ably-labs/react-hooks";
+import styles from "../styles/Home.module.css";
+import Image from "next/image";
+import { formatDate } from "../lib/dates";
+
+/* Subscribes to headline messages from the "news-list" channel
+and provides a form to enter new headlines which it publishes to 
+the same channel */
+export default function Headlines(props) {
+  let inputBox = null;
+  let messageEnd = null;
+
+  const [headlineText, setHeadlineText] = useState("");
+  const [headlines, updateHeadlines] = useState(props.history);
+  const headlineTextIsEmpty = headlineText.trim().length === 0;
+
+  const [channel, ably] = useChannel("headlines", (headline) => {
+    updateHeadlines((prev) => [...prev, headline]);
+  });
+
+  /* Process each message to retrieve the timestamp and author (client Id) */
+  const HeadlinePreviews = ({ items }) => {
+    const previews = items.map((headline, index) => {
+      let author = "";
```

Comment:
The sanitisation of the data should be done in the component that controls the state, in this case, the headlines component. This would leave the headlinePreviews component to be a render-only component, which is easier to read/understand.

You could probably extract the data sanitisation into its own function, and then call it when the useChannel callback is invoked.